src/protocol/secureRequests.ts

```diff
@@ -0,0 +1,389 @@
+ */
+function serializeSecureRequestMsg(
+  client: Client,
+  msgId: Buffer,
+  secureRequestType: LatticeSecureMsgType,
+  payloadData: Buffer,
+): Buffer {
+  // Sanity check request data
+  if (msgId.length !== 4) {
+    throw new Error('msgId must be four bytes');
+  }
+  const isValidRequestType = (
+    secureRequestType === LatticeSecureMsgType.connect ||
+    secureRequestType === LatticeSecureMsgType.encrypted
+  );
+  const isValidConnectPayloadDataSz = (
+    payloadData.length === szs.connect.request.data  
+  );
+  const isValidEncryptedPayloadDataSz = (
+    payloadData.length === szs.requestPayloadData  
+  );
+  const isValidPayloadDataSz = (
+    isValidConnectPayloadDataSz || isValidEncryptedPayloadDataSz
+  ); 
+  if (!isValidRequestType) {
+    throw new Error('Invalid Lattice secure request type');
+  } else if (!isValidPayloadDataSz) {
+    throw new Error('Invalid Lattice secure request payload size');
+  }
+
+  // Build payload
+  let payload;
+  const _payload = Buffer.alloc(szs.requestPayload);
+  payload[0] = secureRequestType;
+  payloadData.copy(payload, 1);
+
+  // Encrypt if needed
+  if (isValidEncryptedPayloadDataSz) {
+    payload = aes256_encrypt(_payload, client.sharedSecret);
+  } else {
+    payload = _payload;
+  }
+
+  // Construct the object first
+  const req: LatticeSecureRequest = {
+    protocolVersion: LatticeProtocolVersion.v1,
+    msgType: LatticeMsgType.secure,
+    msgId,
+    payloadLen: payload.length,
+    payload,
+    checksum: checksum(payload),
+  };
+
+  // Now serialize the whole message
+  const msg = Buffer.alloc(szs.requestMsg);
+  let off = 0;
+  // Header
+  msg.writeUInt8(req.protocolVersion, off);
+  off += 1;
+  msg.writeUInt8(req.msgType, off);
+  off += 1;
+  req.msgId.copy(msg, off);
+  off += req.msgId.length;
+  msg.writeUInt16LE(req.payloadLen, off);
+  off += 2;
+  // Payload
+  msg.writeUInt8(req.payload.requestType, off);
+  off += 1;
+  req.payload.data.copy(msg, off);
+  off += req.payload.data.length;
+  // Footer
+  msg.writeUInt32LE(req.checksum, off);
+  off += 4;
+
+  // We have our serialized secure message!
+  return msg;
+}
+
+/**
+ * @internal
+ * Deserialize a Lattice response and get the payload data
+ * @param client - Instance of Client
+ * @param msgId - 4 byte ID from the request; should match response
+ * @param msg - Buffer received from the Lattice
+ * @return {Buffer} 1696 bytes of payload data (may be encrypted)
+ */
+function deserializeResponseMsgPayloadData(
+  client: Client,
+  msgId: Buffer,
+  msg: Buffer,
+): Buffer {
+  // Sanity check on total message size
+  if (msg.length !== szs.responseMsg) {
+    throw new Error('Wrong Lattice response message size');
+  }
+
+  // Deserialize the message
+  const checksumOffset = 9 + szs.responsePayloadData;
+  const resp: LatticeSecureResponse = {
+    protocolVersion: msg.readUInt8(0),
+    msgType: msg.readUInt8(1),
+    msgId: msg.slice(2, 6),
+    payloadLen: msg.readUInt16LE(6),
```

Comment:
For clarity, we should define another type called `LatticeMessageHeader` with just these four values.